### PR TITLE
allow user to request rabbitmq vhost

### PR DESCRIPTION
**Pull request by miltontony**
- [x] Allow user to request vhost in UI
- [x] Create vhost + use using rabbitmq API
- [x] Ensure `Vhost Name` is a required field if `Vhost neeeded` is ticked
- [x] Hypothesis Tests
- [x] Tests

### CODE REVIEW

File: mc2/controllers/base/models.py [hunk 1]
```diff
@@ -169,6 +180,19 @@ def get_marathon_app_data(self):
...
+        if self.rabbitmq_vhost_needed and self.rabbitmq_vhost_name:
+            self.rabbitmq_manager.create_rabbitmq_vhost()
+            envs.update({
+                'BROKER_URL':
+                    'amqp://%(username)s:%(password)s@%(host)s//%(name)s' %
+                    {
+                        'username': self.rabbitmq_vhost_username,
+                        'password': self.rabbitmq_vhost_password,
+                        'host': self.rabbitmq_vhost_host,
+                        'name': self.rabbitmq_vhost_name,
```
Comment (smn) [inline mc2/controllers/base/models.py]: this may need to be urlencoded

**Comment by miltontony** [comment]
@smn no, I'm not introducing the Resource model just yet - we agreed to keep things simple for now. For celery, you'd simply set the ENV variable for the VHOST from the values you get from the APP.

**Comment by smn** [comment]
I'm sure you're aware of the cognitive overhead of trying to keep these env vars updated everywhere so :+1:

**Comment by miltontony** [comment]
@smn yes, the agreement was that this implementation is no worse than the current way which also uses ENV vars.

**Comment by smn** [comment]
:+1: let's merge?

**Comment by miltontony** [comment]
@smn now ready for review ;)

**Comment by smn** [comment]
:+1: @miltontony 🍰

**Pull request merged**